src/Workspaces.cc

```diff
@@ -389,6 +391,19 @@ Workspaces::insert(PWinObj *wo, bool raise)
                 // window in the next layer.
                 if ((*it)->getLayer() > wo->getLayer()) {
                     top_obj = *it;
+                    if (iconify_fullscreen) {
+                        for (iterator it2 = it; it2 != _wobjs.end(); ++it2) {
+                            if ((*it2)->getWorkspace() != wo->getWorkspace()) {
+                                continue;
+                            }
+                            if ((*it2)->getLayer() <= LAYER_DOCK) {
+                                continue;
+                            }
+                            if ((*it2)->isMapped() && (*it2)->isFullscreen()) { // FIXME maximized
```

Comment:
One flaw I just noticed is this assumption that fullscreen window covers full screen (and hides everything) is wrong, because you _can_ move fullscreen windows in pekwm!

Reply:
I still think it's a fair assumption to make as they are "intended" to be taking up the complete screen. Or, do as previously suggested remove fullscreen state if the window is moved.